### PR TITLE
sketch supermorecado

### PR DESCRIPTION
**Pull request by vincentsarago**
ref https://github.com/developmentseed/cogeo-mosaic/discussions/202

Note: There will be a long way to go for full TMS support

### CODE REVIEW

File: cogeo_mosaic/supermorecado.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
+"""Supermercado.burntiles but for other TMS."""
```
Comment (vincentsarago) [inline cogeo_mosaic/supermorecado.py:1]: add not about provenance of the code and License

File: cogeo_mosaic/supermorecado.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    """Get bounds extrema for a feature."""
+    if geometry["type"] == "Polygon":
+        x, y = zip(*[c for part in geometry["coordinates"] for c in part])
+
+    elif geometry["type"] == "LineString":
+        x, y = zip(*[c for c in geometry["coordinates"]])
+
+    elif geometry["type"] == "Point":
+        x, y = geometry["coordinates"]
+        return x, y, x, y
```
Comment (vincentsarago) [inline cogeo_mosaic/supermorecado.py:23]: we should bring https://github.com/mapbox/supermercado/pull/50 over

File: cogeo_mosaic/supermorecado.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+@attr.s
+class burnTiles:
+    """Burntiles."""
+
+    tms: morecantile.TileMatrixSet = attr.ib(default=WEB_MERCATOR_TMS)
+
+    @tms.validator
+    def _check_for_quadtree_support(self, attribute, value: morecantile.TileMatrixSet):
+        if not value._is_quadtree:
+            raise ValueError(f"{value.identifier} TMS does not support quadtree.")
```
Comment (AndrewAnnex) [inline cogeo_mosaic/supermorecado.py:86]: supporting more than just quadtree structures would be my hope here as equirectangular codes for planetcantile current use 2,1 matrix scales. Is there a limitation of the burn code I'm not seeing here that excludes other scales?

Comment (vincentsarago) [reply cogeo_mosaic/supermorecado.py:86]: you're right quadkey support is only needed for mosaic, not for burntiles 👍

File: cogeo_mosaic/supermorecado.py [hunk 4]
```diff
@@ -145,9 +151,11 @@ def make_transform(self, tilerange: Dict, zoom: int) -> Affine:
 
         return Affine(xcell, 0, ulx, 0, -ycell, uly)
 
-    def burn(self, polys: Sequence[Dict[Any, Any]], zoom: int) -> numpy.ndarray:
+    def burn(
+        self, polys: Sequence[Dict[Any, Any]], zoom: int, truncate: bool = True
+    ) -> numpy.ndarray:
         """Burn geometries to Tiles."""
-        bounds = find_extrema(polys)
+        bounds = self.find_extrema(polys, truncate=truncate)
```
Comment (vincentsarago) [inline cogeo_mosaic/supermorecado.py:158]: I have added a `truncate` options because for sometimes you could want to create tiles outside the tms bbox

**Comment by vincentsarago** [comment]
@AndrewAnnex I'll wait for your 👍 to merge this then we can look at the other part of the code that will need a change

**Comment by AndrewAnnex** [comment]
I think it looks great, haven’t tried to test it with titiler yet though as some updates are needed there to accept other TMSs.- Dr. Andrew AnnexOn Mar 13, 2023, at 2:47 AM, Vincent Sarago ***@***.***> wrote:﻿
@AndrewAnnex I'll wait for your 👍 to merge this then we can look at the other part of the code that will need a change

**Comment by vincentsarago** [comment]
>  with titiler yet though as some updates are needed there to accept other TMSs. 

Yes a lot needs to happen for this

**Review by AndrewAnnex — APPROVED** [review]

**Comment by jlaura** [comment]
This is awesome to see. What is the best way to assist on this? I see that reviews are good to go and tests are in for WebMercator. Is it appropriate to test with another TMS and start to see where else breakages occur?

**Comment by vincentsarago** [comment]
👋 sorry I've been busy on other projects.
I think this PR is good to go, but yeah adding test for other TMS would be great 🤷 but this can be in another PR later

**Pull request merged**